### PR TITLE
1706: Add configuration box to hide author message

### LINKED ISSUE #1706: For some games the author of a message shouldn't be visible
For one wargame we're simulating spoken radio comms via a chat channel.

In spoken comms the identity of the person sending the message will not be known to the recipients.

So, we should have a flag in game-admin over to "Hide identity of message authors" in some chat channels.  So, this will mean introducing a "options" panel at the top of "Chat" rooms - like we do for `mapping` and `collab status board`.

Start off with the backend data changes (Ian), then the admin page changes, then the chat channel itself.

### PR DESCRIPTION
**Pull request by lebaphi**
Fixes #1706 

https://user-images.githubusercontent.com/26705621/183231050-e814efc9-7e4c-4af5-8f20-735beb828d11.mov

### CODE REVIEW

**Comment by IanMayo** [comment]
Perfect.  Good video @lebaphi :-)

File: packages/components/src/local/organisms/setting-channels/index.tsx [hunk 1]
```diff
@@ -77,6 +77,7 @@ export const SettingChannels: React.FC<PropTypes> = ({
   const [participantKey, confirmRemoveParticipant] = useState<number>(-1)
   const [postRemoveActionConfirmed, setPostRemoveActionConfirmed] = useState<boolean>(false)
   const [problems, setProblems] = useState<string>('')
+  const [isChecked, setChecked] = useState<boolean>(false)
```
Comment (IanMayo) [inline packages/components/src/local/organisms/setting-channels/index.tsx]: 'Checked' is a bit vague. Could we use 'isChatAuthorHidden` and `setChatAuthorHidden` instead please?

File: packages/client/src/Components/ChatChannel.tsx [hunk 2]
```diff
@@ -88,8 +87,8 @@ const ChatChannel: React.FC<{ channelId: string, isCustomChannel?: boolean }> =
   return (
     <div className={channelTabClass} data-channel-id={channelId}>
       {
-        typeof isCustomChannel === 'boolean' && isCustomChannel
-          ? <ChannelMessagesList 
+        typeof isCustomChannel === 'boolean' && !isCustomChannel
```
Comment (IanMayo) [inline packages/client/src/Components/ChatChannel.tsx]: this new `!` looks scary.  I'm pretty sure `ChannelMessagesList` is the component to use for `isCustomChannel`.  

Will double-check code when I get chance.

Comment (lebaphi) [reply packages/client/src/Components/ChatChannel.tsx]: sorry @IanMayo , it's a typo, I updated

**Review by IanMayo — APPROVED** [review]
Great, thanks.

**Pull request merged**